app/questionnaire/navigator.py

```diff
@@ -59,10 +59,9 @@ def build_path(self, blocks, this_location):
         """
         path = []
         block_index = 0
-        blocks_len = len(blocks)
 
         # Keep going unless we've hit the last block
-        while block_index < blocks_len:
+        while block_index < len(blocks):
```

Comment:
Eh really? I would think that length was only ever evaluated once. Happy to revert if it does save time..